feature/system/tests/system_base_test/g_protocol_test.go

```diff
@@ -0,0 +1,184 @@
+	}
+	return fmt.Sprintf("%s:%d", s.GetOutsideIp(), s.GetOutside())
+
+}
+
+type rpcCredentials struct {
+	*creds.UserPass
+}
+
+func (r *rpcCredentials) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
+	return map[string]string{
+		"username": "admin",
+		"password": "admin",
+	}, nil
+}
+
+func (r *rpcCredentials) RequireTransportSecurity() bool {
+	return true
+}
+
+// TestGNMIClient validates that the DUT listens on standard gNMI Port.
+func TestGNMIClient(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	target := resolveService(t, dut, "gnmi", 9339)
+	credOpts := []grpc.DialOption{grpc.WithBlock(), grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{InsecureSkipVerify: true}))} // NOLINT
+	creds := &rpcCredentials{}
+	credOpts = append(credOpts, grpc.WithPerRPCCredentials(creds))
+	t.Logf("gNMI standard port test: %q", target)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, target, credOpts...)
+	if err != nil {
+		t.Fatalf("grpc.Dial failed to: %q", target)
+	}
+	c := gpb.NewGNMIClient(conn)
+	if _, err := c.Get(context.Background(), &gpb.GetRequest{}); err != nil {
+		t.Fatalf("gnmi.Get failed: %v", err)
+	}
+}
+
+// TestGNOIClient validates that the DUT listens on standard gNMI Port.
+func TestGNOIClient(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	target := resolveService(t, dut, "gnoi", 9339)
+	credOpts := []grpc.DialOption{grpc.WithBlock(), grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{InsecureSkipVerify: true}))} // NOLINT
+	creds := &rpcCredentials{}
+	credOpts = append(credOpts, grpc.WithPerRPCCredentials(creds))
+
+	t.Logf("gNOI standard port test: %q", target)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, target, credOpts...)
+	if err != nil {
+		t.Fatalf("grpc.Dial failed to: %q", target)
+	}
+	c := spb.NewSystemClient(conn)
+	_, err = c.Ping(context.Background(), &spb.PingRequest{})
+	if err != nil {
+		t.Fatalf("gnoi.system.Time failed: %v", err)
+	}
+}
+
+// TestGNSIClient validates that the DUT listens on standard gNMI Port.
+func TestGNSIClient(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	target := resolveService(t, dut, "gnsi", 9339)
+	credOpts := []grpc.DialOption{grpc.WithBlock(), grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{InsecureSkipVerify: true}))} // NOLINT
+	creds := &rpcCredentials{}
+	credOpts = append(credOpts, grpc.WithPerRPCCredentials(creds))
+
+	t.Logf("gNSI standard port test: %q", target)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, target, credOpts...)
+	if err != nil {
+		t.Fatalf("grpc.Dial failed to: %q", target)
+	}
+	c := authzpb.NewAuthzClient(conn)
+	_, err = c.Get(context.Background(), &authzpb.GetRequest{})
```

Comment:
Same suggestion as above.

Reply:
i want to leave it like this for now since we will likely want to deal with client to improve validation in the future